src/test/java/com/eleks/academy/whoami/core/state/AbstractGameStateTest.java

```diff
@@ -0,0 +1,77 @@
+package com.eleks.academy.whoami.core.state;
+
+import com.eleks.academy.whoami.core.SynchronousPlayer;
+import com.eleks.academy.whoami.core.impl.PersistentPlayer;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class AbstractGameStateTest {
+
+	@Test
+	void findPlayerTest() {
+		String player = "player";
+		int maxPlayer = 4;
+		final Map<String, SynchronousPlayer> players = new HashMap<>(maxPlayer);
+
+		var state = new StateTest(1, maxPlayer, players);
+
+		players.put(player, new PersistentPlayer(player));
+
+		var findPlayer = state.findPlayer(player);
+		var newPlayer = new PersistentPlayer(player);
+		Optional<SynchronousPlayer> expectedPlayer = Optional.of(newPlayer);
```

Comment:
easier to get from optional than to wrap something into optional

Reply:
Resolved